addons/udes_sale/models/sale_order.py

```diff
@@ -0,0 +1,21 @@
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    NON_NEGATIVE_FIELDS = [
+        "price_unit",
+        "product_uom_qty",
+    ]
+
+    @api.constrains(*NON_NEGATIVE_FIELDS)
+    def _constrain_non_negative_values(self):
```

Comment:
self nit; duplicated code in 3 files

Reply:
Can we add the constrain in models ? 
and NON_NEGATIVE_FIELDS = []
In all classes the specific NON_NEGATIVE_FIELDS ?

Reply:
Good idea, we probably can do something generic. Let's create an issue to properly design this, since it could be a property on the field too.

Reply:
i think we still need to define the `@api.constrains(...)` on every model, but we could at least split out the check into a basemodel function

Reply:
https://unipart.atlassian.net/browse/SE-1824 raised